### PR TITLE
TINY-7785: Use `navigator.userAgentData` for browser detection when supported

### PR DESCRIPTION
**Pull request by ltrouton**
Related Ticket: TINY-7785

Description of Changes:
* Use new userAgentData API to detect browser type and major version in Sand

Pre-checks:
* [X] Changelog entry added
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [X] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/sand/CHANGELOG.md [hunk 1]
```diff
@@ -8,3 +8,4 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 - Upgraded to Katamari 8.0, which is incompatible with Katamari 7.0 if used in the same bundle.
+- Detection logic has been updated to use `navigator.userAgentData` for browser detection if it is available.
```
Comment (lnewson) [inline modules/sand/CHANGELOG.md:11]: This maybe should be an improvement? Not sure, what do others think?

File: modules/sand/src/main/ts/ephox/sand/detect/UaData.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const detectBrowser = (browsers: PlatformInfo[], userAgentData: UserAgentData): Optional<UaInfo> => {
+  return Arr.findMap<UserAgentDataBrand, UaInfo>(userAgentData.brands, (uaBrand) => {
+    const lcBrand = uaBrand.brand.toLowerCase();
+    return Arr.find(browsers, (browser) => lcBrand === browser.brand?.toLowerCase())
+      .map((info) => {
+        return {
+          current: info.name,
+          version: Version.nu(parseInt(uaBrand.version, 10), 0)
+        };
+      });
```
Comment (lnewson) [inline modules/sand/src/main/ts/ephox/sand/detect/UaData.ts:28]:
```suggestion
    return Arr.find(browsers, (browser) => lcBrand === browser.brand?.toLowerCase()).map((info) => ({
      current: info.name,
      version: Version.nu(parseInt(uaBrand.version, 10), 0)
    }));
```

File: modules/sand/src/main/ts/ephox/sand/detect/UaData.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  });
+};
+
+export const UaData = {
```
Comment (lnewson) [inline modules/sand/src/main/ts/ephox/sand/detect/UaData.ts:32]: We're trying to move away from this format, as it doesn't treeshake well. Can we just do something like this instead?
```
export {
  detectBrowser
}
```

**Review by lnewson — APPROVED** [review]
Can we also please add a TinyMCE changelog?

File: modules/sand/CHANGELOG.md (hunk 1 above)
Comment (ltrouton) [reply modules/sand/CHANGELOG.md:11]: Yep, happy to move it

**Review by JamesToohey — APPROVED** [review]

File: modules/sand/src/main/ts/ephox/sand/detect/UaData.ts (hunk 3 above)
Comment (ltrouton) [reply modules/sand/src/main/ts/ephox/sand/detect/UaData.ts]: Changed `UaData` and `UaString`

**Review by diogo405 — APPROVED** [review]

**Pull request merged**